src/components/FeatureViews/Accounts/TabComponents/InstructorSchedule.js

```diff
@@ -1,130 +1,124 @@
-					if (all_day) {
-						endDate.setDate(endDate.getDate() + 1);
-					}
-					return {
-						allDay: all_day,
-						end: endDate,
-						start,
-						title: description || "Out of Office",
-					};
-				}
-				)
-				: [],
-		[instructor]
-	);
 
-	const allDayOOO = useMemo(() => OOO.some(({allDay}) => allDay), [OOO]);
-
-	const hoursWorked = useMemo(
-		() =>
-			Math.round(
-				Object.values(sessions[instructorID] || []).reduce((hours, session) => {
-					const start = new Date(session.start_datetime);
-					const end = new Date(session.end_datetime);
-					if (start > Date.now()) {
-						return hours;
-					} else if (end < Date.now()) {
-						return hours + toHours(end - start);
-					}
-					return hours + toHours(end - Date.now());
-				}, 0) * 100
-			) / 100,
-		[sessions, instructorID]
-	);
+	const { loading, error, data } = useQuery(GET_INSTRUCTOR_INFO, {
+		variables: {instructorID}
+	})
 
-	const instructorBusinessHours = useMemo(
-		() =>
-			instructor
-				? Object.values(instructor.schedule.work_hours).map(
-				({day, start, end}) => ({
-					daysOfWeek: [day],
-					endTime: end,
-					startTime: start,
-				})
-				)
-				: [],
-		[instructor]
-	);
+	if (loading ) return null;
+
+	if (error ) return "Unable to load schedule";
 
-	if (
-		teachingSessions.length === 0 &&
-		instructorBusinessHours.length === 0 &&
-		OOO.length === 0
-	) {
-		if (
-			hooks.isLoading(
-				tutoringEnrollmentStatus,
-				OOOstatus,
-				classEnrollmentStatus,
-				availabilityStatus
-			)
-		) {
-			return <Loading/>;
-		} else if (
-			hooks.isFail(
-				tutoringEnrollmentStatus,
-				OOOstatus,
-				classEnrollmentStatus,
-				availabilityStatus
-			)
-		) {
-			return "Unable to load schedule!";
+	const { user } = data.instructor;
+
+	const today = new Date(Date.now());
+
+	const hoursWorkedThisMonth = parseInt(user.instructor.sessionSet.reduce((hours, session) => {
+		const start = new Date(session.startDatetime);
+		const end = new Date(session.endDatetime);
+
+		if (start > Date.now()) {
+			return hours;
 		}
-	}
+		else if (start.getMonth() === today.getMonth() &&
+			start.getFullYear() === today.getFullYear()) {
+			return hours + toHours(end - start);
+		} else {
+			return hours;
+		}
+
+	}, 0));
 
+	const instructorBusinessHours = user.instructor.instructoravailabilitySet.map(({dayOfWeek, startTime, endTime}) => ({
+		dayOfWeek: [dayOfWeek],
+		startTime, 
+		endTime
+	}))
+
+	const teachingSessions = user.instructor.sessionSet.map(({endDatetime, startDatetime, title}) => ({
+		title: title,
+		start: new Date(startDatetime),
+		end: new Date(endDatetime)
+	}));
+
+	let allDayOOO = false;
+	// Out Of Office
+	const OOO = user.instructor.instructoroutofofficeSet.map(({startDatetime, endDatetime, description}) => {
+		let start;
+		let allDay = false;
+		// Checks to see if the start and end time span the whole day
+		if (toHours(new Date(startDatetime) - new Date(endDatetime)) >= 24 ) {
```

Comment:
use moment instead of new Date. moment has a .diff function you can use https://momentjs.com/docs/#/displaying/difference/